rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/activity/CovidPatientRegisterActivityTest.java

```diff
@@ -72,15 +77,15 @@ public void testLanguageSwitcherDialogShouldSetCorrectLocale() throws Exception
 
     private void verifyLocaleIsSaved(ListView listView, int position, String locale) {
         listView.performItemClick(null, position, listView.getItemIdAtPosition(position));
-        Assert.assertEquals(locale, LangUtils.getLanguage(RDTApplication.getInstance()));
-        Assert.assertEquals(locale, RDTApplication.getInstance().getResources().getConfiguration().locale.getLanguage());
+        Assert.assertEquals(locale, new Locale(LangUtils.getLanguage(covidPatientRegisterActivity)).getLanguage());
     }
 
     @Test
     public void testRegisterLanguageSwitcherShouldVerifyLocaleIndex() throws Exception {
-        verifySavedLanguageIndex(1);
-        LangUtils.saveLanguage(covidPatientRegisterActivity, locales[0]);
         verifySavedLanguageIndex(0);
+        LangUtils.saveLanguage(covidPatientRegisterActivity, BuildConfig.LOCALE);
+        Utils.updateLocale(covidPatientRegisterActivity);
+        verifySavedLanguageIndex(1);
```

Comment:
What does this test?

Reply:
if we call a method `Langutils.saveLanguage` inside this method it does update the `resources.updateConfiguration(configuration, resources.getDisplayMetrics());` that's why i have called the method `Utils.updateLocale`.

If we call a method `RDTApplication.getInstance().getResources().getConfiguration().locale.getLanguage()` after `LangUtils.saveLanguage` it returns a previous locale not updated locale.